Selenium_WebDriver/FinalTaskTests.cs

```diff
@@ -0,0 +1,186 @@
+        List<string> _productsAddToWishlists = new List<string>() { "Blouse" };
+
+        [TearDown]
+        public void CleanupCreatedData()
+        {
+            TakeScreenshot();
+
+            HeaderPage headerPage = new HeaderPage(driver);
+            headerPage.GoToMyAccount();
+
+            MyAccountPage myAccountPage = new MyAccountPage(driver);
+            myAccountPage.GoToMyWishlist();
+
+            MyWishlistPage myWishlistPage = new MyWishlistPage(driver);
+            myWishlistPage.RemoveWishlists();
+
+            headerPage.SingOut();            
+        }
+
+        [AllureName("Verifies create account")]
+        [TestCase(FIRST_NAME, LAST_NAME, PASSWORD, ADDRESS, CITY, STATE, ZIP_CODE, COUNTRY, PHONE)]
+        public void CreateAccountTest(string fName, string lName, string password, string address, string city, string state, string zipCode, string country, string phone)
+        {
+            AuthenticationPage authenticationPage = new AuthenticationPage(driver);
+            authenticationPage.GoToUrl();
+            string email = authenticationPage.GetRandomEmail();
+            authenticationPage.CreateAccount(email);
+
+            Assert.IsTrue(authenticationPage.IsAccountCreated(), "{0} is invalid email address", email);
+
+            authenticationPage.Registration(fName, lName, password, address, city, state, zipCode, country, phone);
+
+            MyAccountPage myAccountPage = new MyAccountPage(driver);
+
+            Assert.IsTrue(myAccountPage.IsAccountRegisteredOrSignedIn(), "Required fields are not filled");
+        }
+
+        [AllureName("Verifies LogIn")]
+        [Test]
+        public void SignInTest()
+        {
+            AuthenticationPage authenticationPage = new AuthenticationPage(driver);
+            authenticationPage.GoToUrl();
+            authenticationPage.SignIn(EMAIL, PASSWORD);
+
+            MyAccountPage myAccountPage = new MyAccountPage(driver);
+
+            Assert.IsTrue(myAccountPage.IsAccountRegisteredOrSignedIn(), "Invalid email address {0} or password", EMAIL);
+        }
+
+        [AllureName("Verifies add to wishlists")]
+        [Test]
+        public void AutoCreateWishlistTest()
+        {
+            AuthenticationPage authenticationPage = new AuthenticationPage(driver);
+            authenticationPage.GoToUrl();
+            authenticationPage.SignIn(EMAIL, PASSWORD);
+
+            MyAccountPage myAccountPage = new MyAccountPage(driver);
+            myAccountPage.GoToMyWishlist();
+
+            MyWishlistPage myWishlistPage = new MyWishlistPage(driver);
+
+            Assert.IsFalse(myWishlistPage.IsWishlistExist(), "Wishlist has already created");
+
+            myWishlistPage.GoToHomePage();
+
+            HomePage homePage = new HomePage(driver);
+            homePage.OpenProduct(PRODUCT_NAME);
+
+            if (homePage.IsFrameOpened())
```

Comment:
if-else constructions should be excluded from tests. Generally, it's a bad practice, so it would be glad to handle current case at the Pages level. In critical case, just wrap it into a separate method at the tests level and call it in tests directly.
The same point regarding foreach in one of your bellow tests:)

Reply:
done, thanks